core/tracker/inclusion.go

```diff
@@ -423,27 +425,33 @@ func checkAttestationInclusion(sub submission, block block) (bool, error) {
 
 // checkAttestationV2Inclusion checks whether the attestation is included in the block.
 func checkAttestationV2Inclusion(sub submission, block blockV2) (bool, error) {
+	subData, ok := sub.Data.(core.VersionedAttestation)
+	if !ok {
+		return false, errors.New("invalid attestation")
+	}
+
 	att, ok := block.AttestationsByDataRoot[sub.AttDataRoot]
 	if !ok {
 		return false, nil
 	}
-	attAggBits, err := att.AggregationBits()
-	if err != nil {
-		return false, errors.Wrap(err, "get attestation aggregation bits")
+
+	var attesterDutyData *eth2v1.AttesterDuty
+	for _, ad := range block.AttDuties {
+		if *subData.ValidatorIndex == ad.ValidatorIndex {
+			attesterDutyData = ad
+			break
+		}
 	}
 
-	subData, ok := sub.Data.(core.VersionedAttestation)
-	if !ok {
-		return false, errors.New("invalid attestation")
+	if attesterDutyData == nil {
+		return false, nil
```

Comment:
can we add a debug output here that we did not find a matching validator as there should be no such the case I guess

Reply:
Practically we cannot end up in this scenario I believe. It's just a sanity check for the pointer. I will refactor it not to be a pointer